### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by dbogdan2000**
Added unit tests for TaskRepository and DirectionRepository.

### CODE REVIEW

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+{
+    public class DirectionRepositoryTest
+    {
+        [Fact]
+        public async ThreadTask GetAll_OK()
+        {
+            //Arrange
+            await using var context = ContextCreator.CreateContext();
+            var dbSet = context.Set<Direction>();
+            var directionRepository = new DirectionRepository(context);
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs:21]: Let's refactor it. Move whole necessary initialization in separate fixture as I showed you.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+                                             {
+                                                 Name = "Name3",
+                                                 Description = "Description3",
+                                             },
+                                             new()
+                                             {
+                                                 Name = "Name4",
+                                                 Description = "Description4",
+                                             }
+                                         };
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs:44]: Move data seeding in fixture too

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+            await dbSet.AddRangeAsync(directions);
+            await context.SaveChangesAsync();
+
+            //Act
+            var result = directionRepository.GetAll();
+
+            //Assert
+            Assert.NotEmpty(result);
+            Assert.Equal(directions.Count, await result.CountAsync());
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs:53]: So when you create separate fixture you have to compare exact count.
Example:
<code>
   Assert.Equal(1, await result.CountAsync())
</code>
You mustn't use context directly.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+            var direction = new Direction()
+                            {
+                                Name = "Name1",
+                                Description = "Description1"
+                            };
+            await dbSet.AddAsync(direction);
+            await context.SaveChangesAsync();
+
+            //Act
+            var result = await directionRepository.GetById(1);
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs:73]: You have to remember Id in fixture class and put it here.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+            var direction = new Direction()
+                            {
+                                Name = "Name1",
+                                Description = "Description1"
+                            };
+            await dbSet.AddAsync(direction);
+            await context.SaveChangesAsync();
+
+            //Act, Assert
+            await Assert.ThrowsAsync<ArgumentException>(() => directionRepository.GetById(2));
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs:95]: Hmm... Why here are you checking ArgumentException? Here should be EntityNotExistException

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+                       };
+
+            //Act
+            var entity = await directionRepository.Create(direction);
+            await context.SaveChangesAsync();
+            var result = directionRepository.GetAll();
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.Equal(await result.CountAsync(), directions.Count + 1);
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs:139]: I don't see the reason to check records count in context. It is not main goal of test for creating.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.Equal(await result.CountAsync(), directions.Count);
+            Assert.Equal(newName,
+                         result.First()
+                               .Name);
+            Assert.Equal(newDescription,
+                         result.First()
+                               .Description);
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs:223]: Again I don't see the reason to check it. Currently you have to chek result from Update method.

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi man!
I did some comments. Let me know if any questions.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+        {
+            
+            //Act, Assert
+            await Assert.ThrowsAsync<ArgumentNullException>(() => _fixture.Repository.Create(null));
+        }
+
+        [Fact]
+        public async ThreadTask Update_OK()
+        {
+
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs:87]: Arrange?

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+        }
+
+        [Fact]
+        public async ThreadTask Delete_OK()
+        {
+            //Act
+             await _fixture.Repository.Delete(_fixture.DeleteDirectionId);
+
+            //Assert
+            Assert.NotNull(await _fixture.Context.Directions.FindAsync(_fixture.DeleteDirectionId));
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/DirectionRepositoryTest.cs:119]: Wait what? Why NotNull?

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Fixtures/BaseRepositoryFixture.cs [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        
+        public void Dispose()
+        {
+            Context.Dispose();
+        }
+
+        protected abstract TRepository CreateRepository();
+        protected abstract void InitDatabase();
+        
+
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Fixtures/BaseRepositoryFixture.cs:32]: Too many emapty spaces

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/TaskRepositoryTest.cs [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+        public async ThreadTask GetById_IncorrectData_Fail()
+        {
+            //Act, Assert
+            await Assert.ThrowsAsync<EntityNotExistException>(() => _fixture.Repository.GetById(5));
+        }
+
+        [Fact]
+        public async ThreadTask Create_OK()
+        {
+            var newTask = new Task
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/TaskRepositoryTest.cs:65]: Arrange?

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/TaskRepositoryTest.cs [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+        public async ThreadTask Create_IncorrectData_Fail()
+        {
+            //Act, Assert
+            await Assert.ThrowsAsync<ArgumentNullException>(() => _fixture.Repository.Create(null));
+        }
+
+        [Fact]
+        public async ThreadTask Update_OK()
+        {
+            var updatedName = "Updated Name";
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/TaskRepositoryTest.cs:98]: Arrange?

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/TaskStateRepositoryTest.cs [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+            
+            //Act, Assert
+            await Assert.ThrowsAsync<ArgumentNullException>(() => _fixture.Repository.Create(null));
+        }
+
+        [Fact]
+        public async ThreadTask Update_OK()
+        {
+            
+            var updatedStateName = "Updated Name";
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/TaskStateRepositoryTest.cs:82]: Arrange

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi!
So it looks like better. Fix pls some little missed comments and I'll approve PR.

**Review by PushMeNow — APPROVED** [review]
GJ! Approved

**Review by ArtyomKolosov2 — APPROVED** [review]
LGTM. Approved!

**Pull request merged**